ddl/ddl_api.go

```diff
@@ -2533,6 +2628,68 @@ func (d *ddl) DropColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTa
 	return errors.Trace(err)
 }
 
+// DropColumns will drop multi-columns from the table, now we don't support drop the column with index covered.
+func (d *ddl) DropColumns(ctx sessionctx.Context, ti ast.Ident, specs []*ast.AlterTableSpec) error {
+	schema, t, err := d.getSchemaAndTableByIdent(ctx, ti)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	tblInfo := t.Meta()
+
+	dropingColumnNames := make(map[string]bool)
+	for _, spec := range specs {
+		if !dropingColumnNames[spec.OldColumnName.Name.L] {
+			dropingColumnNames[spec.OldColumnName.Name.L] = true
+		} else {
+			return errors.Trace(infoschema.ErrColumnExists.GenWithStackByArgs(spec.OldColumnName.Name.O))
+		}
+	}
+	if len(tblInfo.Columns) == len(specs) {
+		return ErrCantRemoveAllFields.GenWithStack("can't drop all columns in table %s",
+			tblInfo.Name)
+	}
+
+	var colNames []model.CIStr
+	for _, spec := range specs {
+		// Check whether dropped column has existed.
+		colName := spec.OldColumnName.Name
+		col := table.FindCol(t.VisibleCols(), colName.L)
+		if col == nil {
+			err = ErrCantDropFieldOrKey.GenWithStack("column %s doesn't exist", colName)
+			if spec.IfExists {
+				ctx.GetSessionVars().StmtCtx.AppendNote(err)
+				continue
+			}
+			return err
+		}
+
+		if err = isDroppableColumn(tblInfo, colName); err != nil {
+			return errors.Trace(err)
+		}
+		// We don't support dropping column with PK handle covered now.
+		if col.IsPKHandleColumn(tblInfo) {
+			return errUnsupportedPKHandle
+		}
```

Comment:
Could we put line 2669-2672 to `isDroppableColumn`? Then `DropColumn` and this function needn't write it twice.

Reply:
Hi, `isDroppableColumn` is used in other file, so I create a new function `checkIsDroppableColumn` to reduce code, which contains 20 line code.